test/integration/tcp_proxy_integration_test.cc

```diff
@@ -69,56 +69,21 @@ void TcpProxyIntegrationTest::initialize() {
-    auto* filter_chain = listener->mutable_filter_chains(0);
-    auto* config_blob = filter_chain->mutable_filters(0)->mutable_typed_config();
-
-    ASSERT_TRUE(config_blob->Is<envoy::extensions::filters::network::tcp_proxy::v3::TcpProxy>());
-    auto tcp_proxy_config =
-        MessageUtil::anyConvert<envoy::extensions::filters::network::tcp_proxy::v3::TcpProxy>(
-            *config_blob);
-
-    auto* access_log = tcp_proxy_config.add_access_log();
-    access_log->set_name("accesslog");
-    envoy::extensions::access_loggers::file::v3::FileAccessLog access_log_config;
-    access_log_config.set_path(access_log_path);
-    access_log_config.mutable_log_format()->mutable_text_format_source()->set_inline_string(params);
-    access_log->mutable_typed_config()->PackFrom(access_log_config);
-    auto* runtime_filter = access_log->mutable_filter()->mutable_runtime_filter();
-    runtime_filter->set_runtime_key("unused-key");
-    auto* percent_sampled = runtime_filter->mutable_percent_sampled();
-    percent_sampled->set_numerator(100);
-    percent_sampled->set_denominator(envoy::type::v3::FractionalPercent::DenominatorType::
-                                         FractionalPercent_DenominatorType_HUNDRED);
-    config_blob->PackFrom(tcp_proxy_config);
-  });
-}
-
-inline std::string setupByteMeterAccessLog(ConfigHelper& config_helper,
-                                           Network::Address::IpVersion& version) {
-  auto file_name =
-      fmt::format("access_log{}{}.txt", version == Network::Address::IpVersion::v4 ? "v4" : "v6",
-                  TestUtility::uniqueFilename());
-  std::string access_log_path = TestEnvironment::temporaryPath(file_name);
-  setListenerAccessLog(config_helper, access_log_path,
-                       "DOWNSTREAM_WIRE_BYTES_SENT=%DOWNSTREAM_WIRE_BYTES_SENT% "
-                       "DOWNSTREAM_WIRE_BYTES_RECEIVED=%DOWNSTREAM_WIRE_BYTES_RECEIVED% "
-                       "UPSTREAM_WIRE_BYTES_SENT=%UPSTREAM_WIRE_BYTES_SENT% "
-                       "UPSTREAM_WIRE_BYTES_RECEIVED=%UPSTREAM_WIRE_BYTES_RECEIVED%\n");
-  return access_log_path;
+inline void setupByteMeterAccessLog(BaseIntegrationTest* base) {
+  base->useListenerAccessLog(
+      "DOWNSTREAM_WIRE_BYTES_SENT=%DOWNSTREAM_WIRE_BYTES_SENT% "
+      "DOWNSTREAM_WIRE_BYTES_RECEIVED=%DOWNSTREAM_WIRE_BYTES_RECEIVED% "
+      "UPSTREAM_WIRE_BYTES_SENT=%UPSTREAM_WIRE_BYTES_SENT% "
+      "UPSTREAM_WIRE_BYTES_RECEIVED=%UPSTREAM_WIRE_BYTES_RECEIVED%\n");
 }
 
 // Using a set of T, test a consumer function for a true AssertionResult, or
 // timeout.
 template <typename T>
-inline absl::optional<uint64_t> waitForFunction(std::vector<T>& connection_list,
-                                                std::chrono::milliseconds connection_wait_timeout,
-                                                std::function<AssertionResult(T&)>& func) {
+inline absl::optional<uint64_t> waitForFunction(
```

Comment:
This seems like `HttpIntegrationTest::waitForNextUpstreamConnection`, that waits for a Raw connection instead, perhaps we can turn it into `waitForNextRawConnection()` since that's how it's being used, and push it to the `BaseIntegrationTest` class